clarity/src/vm/analysis/type_checker/v2_1/mod.rs

```diff
@@ -1270,11 +1270,11 @@ impl<'a, 'b> TypeChecker<'a, 'b> {
             if self.epoch.analysis_memory() {
                 let added_memory = u64::from(arg_name.len())
                     .checked_add(arg_type.type_size()?.into())
-                    .ok_or_else(|| CostErrors::CostOverflow)?;
+                    .ok_or(CostErrors::CostOverflow)?;
```

Comment:
I think we want to keep `.ok_or_else(..)` because the closure only runs on error, which means any memory allocations that would happen there would only happen in the error case (instead of eagerly, as would be the case with `.ok_or(..)`).  

Am I remembering that right @jbencin?

Reply:
Yeah, `ok_or()` is eagerly evaluated, which means the value is constructed even if the condition is not triggered, although this may be optimized away on `release` builds, idk for sure

I just looked up the size of `CostErrors` and it's 88 bytes (!) according to Rust analyzer, because it can contain 2 `ExecutionCost` structs, so I think this should be in an `ok_or_else()`

> which means any memory allocations

This isn't `Box`ed so at least there's no heap allocation here